### PR TITLE
[actors] retention actors

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- changes retention people to retention actors and adds group support
*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*
make sure group tests retrieve actors

### CODE REVIEW

File: ee/clickhouse/queries/retention/clickhouse_retention.py [hunk 1]
```diff
@@ -206,138 +195,29 @@ def _get_condition(self, target_entity: Entity, table: str, prepend: str = "") -
             params = {"{}_event".format(prepend): "$pageview"}
         return condition, params
 
-    def _retrieve_people(self, filter: RetentionFilter, team: Team):
```
Comment (macobo) [inline ee/clickhouse/queries/retention/clickhouse_retention.py:209]: Nice!!

File: ee/clickhouse/queries/retention/retention_event_query.py [hunk 2]
```diff
@@ -115,7 +115,10 @@ def get_timestamp_field(self) -> str:
             return f"{self.EVENT_TABLE_ALIAS}.timestamp AS event_date"
 
     def _determine_should_join_distinct_ids(self) -> None:
-        self._should_join_distinct_ids = True
+        if self._filter.aggregation_group_type_index is not None:
```
Comment (macobo) [inline ee/clickhouse/queries/retention/retention_event_query.py:118]: Unrelated optimization right?

File: ee/clickhouse/queries/retention/retention_actors.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+        return (
+            RETENTION_PEOPLE_SQL.format(
+                target_event_query=target_event_query,
+                returning_query=return_query_formatted,
+                filters=prop_filters,
+                actor_field_name=actor_field_name,
+                person_join=""
+                if self.is_aggregating_by_groups
+                else f"JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) pdi on e.distinct_id = pdi.distinct_id",
```
Comment (macobo) [inline ee/clickhouse/queries/retention/retention_actors.py:102]: Please update this with the new method: ee/clickhouse/queries/person_distinct_id_query.py

File: ee/clickhouse/queries/retention/retention_actors.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        target_query, target_params = _get_condition(self._filter.target_entity, table="e")
+        target_query_formatted = f"AND {target_query}"
+        return_query, return_params = _get_condition(self._filter.returning_entity, table="e", prepend="returning")
+        return_query_formatted = f"AND {return_query}"
+
+        actor_field_name = f"{get_aggregation_target_field(self._filter.aggregation_group_type_index, self.EVENT_TABLE_ALIAS, self.DISTINCT_ID_TABLE_ALIAS)} AS actor_id"
+        person_join = (
+            ""
+            if self.is_aggregating_by_groups
+            else f"JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) pdi on e.distinct_id = pdi.distinct_id"
```
Comment (macobo) [inline ee/clickhouse/queries/retention/retention_actors.py:147]: Please update this with the new method: ee/clickhouse/queries/person_distinct_id_query.py

File: ee/clickhouse/queries/retention/clickhouse_retention.py [hunk 5]
```diff
@@ -1,28 +1,18 @@
...
 from ee.clickhouse.models.action import format_action_filter
-from ee.clickhouse.models.person import get_persons_by_uuids
-from ee.clickhouse.models.property import parse_prop_clauses
+from ee.clickhouse.queries.retention.retention_actors import (
+    ClickhouseRetentionActors,
+    ClickhouseRetentionActorsByPeriod,
+)
 from ee.clickhouse.queries.retention.retention_event_query import RetentionEventsQuery
 from ee.clickhouse.queries.util import get_trunc_func_ch
 from ee.clickhouse.sql.person import GET_TEAM_PERSON_DISTINCT_IDS
```
Comment (macobo) [inline ee/clickhouse/queries/retention/clickhouse_retention.py:11]: Dead imports?

File: posthog/queries/test/test_retention.py [hunk 6]
```diff
@@ -422,7 +422,7 @@ def test_retention_people_first_time(self):
             )
 
             self.assertEqual(len(result), 1)
-            self.assertEqual(result[0]["id"], p3.pk)
+            self.assertTrue(result[0]["id"] == p3.pk or result[0]["id"], p3.uuid)
```
Comment (macobo) [inline posthog/queries/test/test_retention.py:425]: Typo here with the comma. This is always truthy?

Maybe extract a `get_actor_id` function in tests instead?

File: ee/clickhouse/queries/retention/retention_actors.py [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+            filters=prop_filters,
+            trunc_func=trunc_func,
+            person_join=person_join,
+            actor_field_name=actor_field_name,
+        )
+
+        date_from = self._filter.date_from + self._filter.selected_interval * self._filter.period_increment
+        date_to = self._filter.date_to
+
+        # NOTE: I'm using `Any` here to avoid typing issues when trying to iterate.
```
Comment (macobo) [inline ee/clickhouse/queries/retention/retention_actors.py:176]: Unclear what this is referring to

**Review by macobo — COMMENTED** [review]
One blocking comment (GET_TEAM_PERSON_DISTINCT_IDS) but otherwise looks good to go! Once updated & rebased feel free to merge

File: posthog/queries/test/test_retention.py (hunk 6 above)
Comment (EDsCODE) [reply posthog/queries/test/test_retention.py]: leaving for now. once we purge postgres stuff these tests will be tightened up

File: ee/clickhouse/queries/retention/retention_event_query.py (hunk 2 above)
Comment (EDsCODE) [reply ee/clickhouse/queries/retention/retention_event_query.py:118]: don't want to join unnecessarily. if person props are needed in group aggregation, the initialization should still set this correctly

**Pull request merged**